### PR TITLE
Codemod: convert mdx to module format

### PR DESCRIPTION
**Pull request by shilman**
Issue: #7101 

Builds on https://github.com/storybookjs/storybook/pull/7418, merge that first before reviewing

## What I did

- Adds `convert-mdx-to-module` codemod + test cases

## How to test

Unit:
```
yarn jest --testPathPattern convert-mdx-to-module.test.js
```

End-to-end:
```
./lib/cli/bin/index.js migrate convert-mdx-to-module --glob "examples/official-storybook/**/*.stories.mdx" --rename mdx:js
```

### CODE REVIEW

File: lib/codemod/README.md [hunk 1]
```diff
@@ -237,3 +239,39 @@ import { Meta, Story } from '@storybook/addon-docs/blocks';
...
+
+For example:
+
+```js
+import React from 'react';
+import Button from './Button';
+import { Meta, Story } from '@storybook/addon-docs/blocks';
+
+<Meta title='Button' />
+
```
Comment (ndelangen) [inline lib/codemod/README.md]: Any markdown here just gets completely dropped?

Comment (shilman) [reply lib/codemod/README.md]: For now, yes

Comment (tmeasday) [reply lib/codemod/README.md]: Hmm.. do you think this is useful if we do that?

Comment (shilman) [reply lib/codemod/README.md]: Absolutely. We're going from a format that supports mixtures of longform documentation and code to one that does not. This completely solves the tedious task of translation in an automated fashion, and for now users will have to figure out themselves how they want to encode that documentation.

If we ever have a recommended way to encode longform docs in module format, we can try to encode that in the codemod.

Comment (tmeasday) [reply lib/codemod/README.md]: I thought you could put markdown in the module format somehow?

Comment (shilman) [reply lib/codemod/README.md]: I don't have a prescription for that. Some users will put it in docgen comments, others in strings, others in `.md` files that get loaded in. When somebody has a good solution they want to push, they can upgrade this codemod accordingly.

I personally think people should be using MDX for longform docs and not module format, and the purpose of this PR is to help people who try it and want to go back for whatever reason.

**Pull request merged**